### PR TITLE
Diesel 1.4.3 (new formula)

### PR DESCRIPTION
**Pull request by zengxs**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Command-line tool for Diesel (Rust ORM library)
<https://github.com/diesel-rs/diesel>

### CODE REVIEW

File: Formula/diesel.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  head "https://github.com/diesel-rs/diesel.git"
+
+  depends_on "rust" => [:build, :test]
+  depends_on "libpq"
+  depends_on "mysql-client"
+  uses_from_macos "sqlite"
+
+  def install
+    system "cargo", "install", "--root", prefix, "--path", "diesel_cli"
```
Comment (chenrui333) [inline Formula/diesel.rb]: can you add `--locked` flag? relates to https://github.com/Homebrew/homebrew-core/issues/46025

Comment (zengxs) [reply Formula/diesel.rb]: There seems to be no `Cargo.lock` file in Diesel's repository.

Comment (chenrui333) [reply Formula/diesel.rb]: Looks like it relates to this issue, https://github.com/diesel-rs/diesel/issues/2209

Historically, it does have `cargo.lock` in the repo though, not sure when and why it got removed.
https://github.com/diesel-rs/diesel/issues/239

**Comment by chenrui333** [comment]
@BrewTestBot test this please

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@zengxs Thanks for your contribution to Homebrew! :tada: 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

File: Formula/diesel.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    system "#{bin}/diesel completions bash > diesel.bash"
+    system "#{bin}/diesel completions zsh > _diesel"
+    system "#{bin}/diesel completions fish > diesel.fish"
+
+    bash_completion.install "diesel.bash"
+    zsh_completion.install "_diesel"
+    fish_completion.install "diesel.fish"
```
Comment (chenrui333) [inline Formula/diesel.rb:23]: cc @SMillerDev

**Pull request closed without merging**